src/main/java/hudson/plugins/checkstyle/CheckStylePublisher.java

```diff
@@ -130,19 +146,28 @@ public String getPattern() {
         return pattern;
     }
 
+    /**
+     * Sets the Ant file-set pattern of files to work with.
+     */
+    @DataBoundSetter
+    public void setPattern(final String pattern) {
+        this.pattern = pattern;
+    }
+
     @Override
     public Action getProjectAction(final AbstractProject<?, ?> project) {
         return new CheckStyleProjectAction(project);
     }
 
     @Override
-    public BuildResult perform(final AbstractBuild<?, ?> build, final PluginLogger logger) throws InterruptedException, IOException {
+    public BuildResult perform(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger) throws InterruptedException, IOException {

```

Comment:
I don't know any plugin extending this one. In any case, the original signature is [inherited from super class](https://github.com/jenkinsci/analysis-core-plugin/pull/31/files#diff-05fdbc9ae8f64389b0960930acf7c276R225) (from the upstream PR), so no need to do it again in any Analysis plugin being adapted.